src/components/Dragons.jsx

```diff
@@ -1,8 +1,15 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchDragonsData } from '../redux/dragons/dragon';
 
 const Dragons = () => {
   const dragonsList = useSelector((state) => state.dragonReducer);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (dragonsList.status !== 'FETCHING_SUCCEEDED') dispatch(fetchDragonsData());
```

Comment:
This is a good way to check if data is present in the list or not. But I think if you call `fetchDragonsData()` in the App component.

Reply:
Thanks Saadat I've implemented the changes.